src/db-service.ts

```diff
@@ -145,12 +155,16 @@ export class CategoryService {
       .then(({ rows }) => rows[0]);
   }
 
-  // Get itemCategories matching given category(s) (intersection)
-  async getItemsByCategories(
+  /**
+   * Get item ids having at least one of the given categories
+   * @param ids category ids an item should have
+   * @param dbHandler
+   * @returns object { id } of items with given categories
+   */
+  async getItemIdsByCategories(
     ids: string[],
     dbHandler: TrxHandler,
   ): Promise<Item[]> {
```

Comment:
Do you mean it's misleading because we get `id` and not `itemId`?

Reply:
No I believe it's misleading because the return type from `getItemIdsByCategories` implies that entire `Item`s will be retrieved. But the SQL query already projects onto the item IDs only with the `SELECT` statement.

Reply:
Ah yes indeed, so with the pick it's clearer 👍